packages/backend/src/models/AIConfig.ts

```diff
@@ -70,15 +84,35 @@ export function parseYamlFile(filepath: string, defaultArch: string): AIConfig {
           contextdir = '.';
         }
 
+        const architectures: string[] = [];
+        if (!('arch' in container)) {
+          architectures.push(defaultArch);
+        } else if (Array.isArray(container['arch']) && container['arch'].every(arch => typeof arch === 'string')) {
+          architectures.push(...container['arch']);
+        } else if (typeof container['arch'] === 'string') {
+          architectures.push(container['arch']);
+        } else {
+          throw new Error('malformed arch property');
+        }
+
+        let contanerfile: string | undefined = undefined;
+        if ('containerfile' in container && isString(container['containerfile'])) {
+          contanerfile = container['containerfile'];
+        }
+
+        if (!('name' in container) || typeof container['name'] !== 'string') {
+          throw new Error('invalid name property: must be string');
+        }
+
         return {
-          arch: Array.isArray(container['arch']) ? container['arch'] : [defaultArch],
-          modelService: container['model-service'] === true,
-          containerfile: isString(container['containerfile']) ? container['containerfile'] : undefined,
+          arch: architectures,
+          modelService: 'model-service' in container && container['model-service'] === true,
+          containerfile: contanerfile,
           contextdir: contextdir,
-          name: assertString(container['name']),
-          gpu_env: Array.isArray(container['gpu-env']) ? container['gpu-env'] : [],
-          ports: Array.isArray(container['ports']) ? container['ports'] : [],
-          image: isString(container['image']) ? container['image'] : undefined,
+          name: container['name'],
+          gpu_env: 'gpu-env' in container && Array.isArray(container['gpu-env']) ? container['gpu-env'] : [],
+          ports: 'ports' in container && Array.isArray(container['ports']) ? container['ports'] : [],
+          image: 'image' in container && isString(container['image']) ? container['image'] : undefined,
```

Comment:
not sure it's covered by tests ?

Reply:
Yeah there was not test for failing cases, only a valid config. I added all the tests needed